src/core/positioning/egenioussreceiver.cpp

```diff
@@ -0,0 +1,128 @@
+#include "egenioussreceiver.h"
+
+#include <QHostAddress>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QJsonValue>
+
+EgenioussReceiver::EgenioussReceiver( const QString &address, const int port, QObject *parent )
+  : AbstractGnssReceiver( parent ), mAddress( address ), mPort( port ), mTcpSocket( new QTcpSocket( this ) )
+{
+  setValid( !mAddress.isEmpty() && mPort > 0 );
+}
+
+void EgenioussReceiver::handleConnectDevice()
+{
+  connect( mTcpSocket, &QTcpSocket::readyRead, this, &EgenioussReceiver::onReadyRead );
+  connect( mTcpSocket, &QTcpSocket::errorOccurred, this, &EgenioussReceiver::onErrorOccurred );
```

Comment:
connect this in the constructor, we should connect/disconnect these every time.

Reply:
Correct :+1: